From 51a6248b4411f06cc1fa0aa5d510ddbda2a85371 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Mon, 30 Sep 2019 10:04:55 +0000
Subject: [PATCH 06/10] ui: ozone: wayland: wayland-ime: Fix keycode conversion

Use the correct way to convert key code to dom code.

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 .../wayland/host/wayland_input_method_context.cc         | 9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/ui/ozone/platform/wayland/host/wayland_input_method_context.cc b/ui/ozone/platform/wayland/host/wayland_input_method_context.cc
index 81a75d466..1fac91c99 100644
--- a/ui/ozone/platform/wayland/host/wayland_input_method_context.cc
+++ b/ui/ozone/platform/wayland/host/wayland_input_method_context.cc
@@ -26,12 +26,6 @@
 
 namespace ui {
 
-namespace {
-
-constexpr int kXkbKeycodeOffset = 8;
-
-}  // namespace
-
 WaylandInputMethodContext::WaylandInputMethodContext(
     WaylandConnection* connection,
     LinuxInputMethodContextDelegate* delegate,
@@ -162,8 +156,7 @@ void WaylandInputMethodContext::OnKeysym(uint32_t key,
   uint8_t flags = 0;  // for now ignore modifiers
   DomKey dom_key = NonPrintableXKeySymToDomKey(key);
   KeyboardCode key_code = NonPrintableDomKeyToKeyboardCode(dom_key);
-  DomCode dom_code =
-      KeycodeConverter::NativeKeycodeToDomCode(key_code + kXkbKeycodeOffset);
+  DomCode dom_code = UsLayoutKeyboardCodeToDomCode(key_code);
   if (dom_code == ui::DomCode::NONE)
     return;
 
-- 
2.20.1

